src/platform/silabs/wifi/SiWx/WifiInterfaceImpl.cpp

```diff
@@ -595,61 +592,32 @@ void WifiInterfaceImpl::ProcessEvent(WiseconnectWifiInterface::WifiPlatformEvent
         JoinWifiNetwork();
         break;
 
-    case WiseconnectWifiInterface::WifiPlatformEvent::kStationDoDhcp:
-        ChipLogDetail(DeviceLayer, "WifiPlatformEvent::kStationDoDhcp");
-        StartDHCPTimer(kDhcpPollIntervalMs);
-        break;
-
-    case WiseconnectWifiInterface::WifiPlatformEvent::kStationDhcpDone:
-        ChipLogDetail(DeviceLayer, "WifiPlatformEvent::kStationDhcpDone");
-        CancelDHCPTimer();
-        break;
-
-    case WiseconnectWifiInterface::WifiPlatformEvent::kStationDhcpPoll:
-        ChipLogDetail(DeviceLayer, "WifiPlatformEvent::kStationDhcpPoll");
-        HandleDHCPPolling();
-        break;
+    case WiseconnectWifiInterface::WifiPlatformEvent::kStationNotify:
+        ChipLogDetail(DeviceLayer, "WifiPlatformEvent::kStationNotify");
+        HandleNotifyConnectivity();
 
     default:
         break;
     }
 }
 
-void WifiInterfaceImpl::HandleDHCPPolling(void)
+void WifiInterfaceImpl::HandleNotifyConnectivity(void)
 {
-    WiseconnectWifiInterface::WifiPlatformEvent event;
-
-    // TODO: Notify the application that the interface is not set up or Chipdie here because we are in an unkonwn state
     struct netif * sta_netif = &wifi_client_context.netif;
-    VerifyOrReturn(sta_netif != nullptr, ChipLogError(DeviceLayer, "HandleDHCPPolling: failed to get STA netif"));
+    VerifyOrReturn(sta_netif != nullptr, ChipLogError(DeviceLayer, "HandleNotifyConnectivity: failed to get STA netif"));
 
 #if (CHIP_DEVICE_CONFIG_ENABLE_IPV4)
-    uint8_t dhcp_state = dhcpclient_poll(sta_netif);
-    if (dhcp_state == DHCP_ADDRESS_ASSIGNED && !mHasNotifiedIPv4)
-    {
-        GotIPv4Address((uint32_t) sta_netif->ip_addr.u_addr.ip4.addr);
-        event = WiseconnectWifiInterface::WifiPlatformEvent::kStationDhcpDone;
-        WiseconnectWifiInterface::PostWifiPlatformEvent(event);
-        NotifyConnectivity();
-    }
-    else if (dhcp_state == DHCP_OFF)
-    {
-        NotifyIPv4Change(false);
-    }
+    GotIPv4Address((uint32_t) sta_netif->ip_addr.u_addr.ip4.addr);
+    NotifyConnectivity();
```

Comment:
Is it necessary to executy the NotifyConnectivity twice in this function?

Reply:
removed